src/App.tsx

```diff
@@ -1,24 +1,275 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useState, useEffect, useRef, useMemo,
+} from 'react';
+import { TodosList } from './components/TodosList';
+import {
+  getTodos, addTodo, removeTodo, updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
 import { UserWarning } from './UserWarning';
+import { Navigation } from './components/Navigation';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoStatus } from './types/TodoStatus';
 
-const USER_ID = 0;
+const USER_ID = 5760;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const newTodoField = useRef<HTMLInputElement>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [messageError, setMessageError] = useState<string>('');
+  const [status, setStatus] = useState<TodoStatus>(TodoStatus.All);
+  const [textField, setTextField] = useState<string>('');
+  const [isProcessing, setIsProcessing] = useState(false);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const handleDelete = (todoId: number) => {
+    setTodos((prev: Todo[]) => prev.filter((todo) => todo.id !== todoId));
+  };
+
+  const handleUpdate = (todoToUpdate: Todo, title?: string) => {
+    setTodos((prev: Todo[]) => prev.map(todo => {
+      if (todo.id === todoToUpdate.id) {
+        updateTodo(USER_ID, todoToUpdate, title);
+
+        const requestBody = title
+          ? { ...todo, title }
+          : { ...todo, completed: !todo.completed };
+
+        return requestBody;
+      }
+
+      return todo;
```

Comment:
you can return from first condition, do it to avoid big nesting of statements
```
if (todo.id !== todoToUpdate.id) {
  return todo
}

// do something

```